external-crates/move/move-compiler/src/sui_mode/typing.rs

```diff
@@ -808,3 +920,124 @@ fn exp(context: &mut Context, e: &T::Exp) {
         _ => (),
     }
 }
+
+fn check_event_emit(context: &mut Context, loc: Loc, mcall: &ModuleCall) {
+    let current_module = context.current_module();
+    let ModuleCall {
+        module,
+        name,
+        type_arguments,
+        ..
+    } = mcall;
+    let Some(first_ty) = type_arguments.first() else {
+        // invalid arity
+        debug_assert!(false, "ICE arity should have been expanded for errors");
+        return;
+    };
+    let is_defined_in_current_module = matches!(first_ty.value.type_name(), Some(sp!(_, TypeName_::ModuleType(m, _))) if m == current_module);
+    if !is_defined_in_current_module {
+        let msg = format!(
+            "Invalid event. The function '{}::{}' must be called with a type defined in the current module, '{}'",
```

Comment:
small nits:
- Current errors around the current module don't seem to list that module
- The ',' between "current module" and the name is stylistically different from existing errors I have seen

Reply:
It brings me great joy to have my error messages finally style checked by someone other than myself (no joke). So fixed :)